jvm_flags.bzl

```diff
@@ -0,0 +1,79 @@
+"""
+buildfarm definitions and configurations around choosing JVM flags for java images.
+"""
+
+SERVER_TELEMETRY_JVM_FLAGS = [
+    "-javaagent:/app/build_buildfarm/opentelemetry-javaagent.jar",
+    "-Dotel.resource.attributes=service.name=server",
+    "-Dotel.exporter.otlp.traces.endpoint=http://otel-collector:4317",
+    "-Dotel.instrumentation.http.capture-headers.client.request",
+    "-Dotel.instrumentation.http.capture-headers.client.response",
+    "-Dotel.instrumentation.http.capture-headers.server.request",
+    "-Dotel.instrumentation.http.capture-headers.server.response",
+]
+
+WORKER_TELEMETRY_JVM_FLAGS = [
+    "-javaagent:/app/build_buildfarm/opentelemetry-javaagent.jar",
+    "-Dotel.resource.attributes=service.name=worker",
+    "-Dotel.exporter.otlp.traces.endpoint=http://otel-collector:4317",
+    "-Dotel.instrumentation.http.capture-headers.client.request",
+    "-Dotel.instrumentation.http.capture-headers.client.response",
+    "-Dotel.instrumentation.http.capture-headers.server.request",
+    "-Dotel.instrumentation.http.capture-headers.server.response",
+]
+
+RECOMMENDED_JVM_FLAGS = [
+    # Enables the JVM to detect if it is running inside a container and automatically adjusts
+    # its behavior to optimize performance. This flag can help ensure that the JVM is
+    # configured optimally for containerized environments.
+    "-XX:+UseContainerSupport",
+
+    # By default, the JVM sets the maximum heap size to 25% of the available memory.
+    # It’s quite conservative.  Let's give the heap more space:
+    "-XX:MaxRAMPercentage=80",
+
+    # The -XX:+UseG1GC flag is recommended for use in Kubernetes environments.
+    # This enables the Garbage-First Garbage Collector, which is optimized for
+    # high-throughput, low-latency applications.  It can help reduce
+    # the memory footprint of the JVM.
+    "-XX:+UseG1GC",
```

Comment:
G1 should be the default in all recent versions of Java. Also there is a newer UseZGC one now but I personally haven't tested it with Buildfarm yet.

Reply:
We should also consider upgrading past java 11.  Do you use anything higher?  We are on 11 simply because its bazel's default.

Reply:
I'm on 19 for a while now. I believe the public images still use 11 though.

Reply:
You're right, `G1GC` is used by default in Java 11.  Removing the flag here.